TestTools/StreamChatTestMockServer/MockServer/MessageResponses.swift

```diff
@@ -534,9 +537,61 @@ public extension StreamMockServer {
+        let messageId = try XCTUnwrap(request.params[EndpointQuery.messageId])
         var json = "{\"\(JSONKey.messages)\":[]}".json
-        let messages = findMessagesByParrentId(messageId)
+        var messages = findMessagesByParrentId(messageId)
+        
+        guard
+            let limitQueryParam = request.queryParams.first(where: { $0.0 == MessagesPagination.CodingKeys.pageSize.rawValue })
+        else {
+            json[JSONKey.messages] = messages
+            return .ok(.json(json))
+        }
+        
+        let limit = (limitQueryParam.1 as NSString).integerValue
+        if let idLt = request.queryParams.first(where: { $0.0 == PaginationParameter.CodingKeys.lessThan.rawValue })?.1 {
+            let messageIndex = messages.firstIndex {
+                idLt == $0[messageKey.id.rawValue] as? String
+            }
+            if let messageIndex = messageIndex {
+                let startWith = messageIndex - limit > 0 ? messageIndex - limit : 0
+                let endWith = messageIndex - 1 > 0 ? messageIndex - 1 : 0
+                messages = Array(messages[startWith...endWith])
+            }
+        } else if let idGt = request.queryParams.first(where: { $0.0 == PaginationParameter.CodingKeys.greaterThan.rawValue })?.1 {
+            let messageIndex = messages.firstIndex {
+                idGt == $0[messageKey.id.rawValue] as? String
+            }
+            if let messageIndex = messageIndex {
+                let messageCount = messages.count - 1
+                let plusLimit = messageIndex + limit
+                let endWith = plusLimit < messageCount ? plusLimit : messageCount
+                messages = Array(messages[messageIndex + 1...endWith])
+            }
+        } else if let idLte = request.queryParams.first(where: { $0.0 == PaginationParameter.CodingKeys.lessThanOrEqual.rawValue })?.1 {
+            let messageIndex = messages.firstIndex {
+                idLte == $0[messageKey.id.rawValue] as? String
+            }
+            if let messageIndex = messageIndex {
+                let minusLimit = messageIndex - limit
+                let startWith = minusLimit > 0 ? minusLimit : 0
+                messages = Array(messages[startWith + 1...messageIndex])
+            }
+        } else if let idGte = request.queryParams.first(where: { $0.0 == PaginationParameter.CodingKeys.greaterThanOrEqual.rawValue })?.1 {
+            let messageIndex = messages.firstIndex {
+                idGte == $0[messageKey.id.rawValue] as? String
+            }
+            if let messageIndex = messageIndex {
+                let messageCount = messages.count - 1
+                let plusLimit = messageIndex + limit
+                let endWith = plusLimit < messageCount ? plusLimit - 1 : messageCount
+                messages = Array(messages[messageIndex...endWith])
+            }
+        } else {
+            messages = Array(messages.suffix(limit))
+        }
```

Comment:
I can see we also have a `limitQuery` function that does exactly this implementation 🤔 

Ideally, we should also change the `limitQuery` to `mockMessages()` which is more intuitive. And then, we should try to extract this logic to a function and reuse it in both places. Something like `fakeMessagePagination()`

WDYT?

Reply:
Done, check it out 🤞